flow/surface_frame.cc

```diff
@@ -26,7 +26,9 @@ SurfaceFrame::SurfaceFrame(sk_sp<SkSurface> surface,
   if (surface_) {
     canvas_ = surface_->getCanvas();
   } else if (display_list_fallback) {
-    dl_recorder_ = sk_make_sp<DisplayListCanvasRecorder>(kGiantRect);
+    FML_DCHECK(!frame_size.isEmpty());
+    dl_recorder_ =
+        sk_make_sp<DisplayListCanvasRecorder>(SkRect::Make(frame_size));
```

Comment:
This right here is what makes the magic for impeller. This code path is used for impeller, and this will cause the SkCanvas implementation underlying the recorder here to `quickReject` drawing ops that fall outside of it, which will gate off a whole lot of allocations of impeller-specific geometry (e.g. not creating `impeller::Path` objects for `SkPath`s that fall entirely outside these bounds).